src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -36,6 +40,21 @@ export class WorkspaceConflictSavedObjectsClientWrapper {
     );
   }
 
+  private isDataSourceType(type: SavedObjectsFindOptions['type']): boolean {
+    if (Array.isArray(type)) {
+      return type.every((item) => item === DATA_SOURCE_SAVED_OBJECT_TYPE);
+    }
+
+    return type === DATA_SOURCE_SAVED_OBJECT_TYPE;
+  }
+  private isConfigType(type: SavedObject['type']): boolean {
+    return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
+  }
+  private formatFindParams(options: SavedObjectsFindOptions): SavedObjectsFindOptions {
+    const isListingDataSource = this.isDataSourceType(options.type);
+    return isListingDataSource ? { ...options, workspaces: null } : options;
```

Comment:
Does the API do something different when `workspaces: null` vs not specify it?

Reply:
No, do you prefer just not specify it?

Reply:
Changed to a cleaner version by not specifying workspaces.